### ISSUE TITLE
Set proxy from QGIS options

### ISSUE THREAD

**Issue by fxprunayre** [INSPIRE-MIF/gml_application_schema_toolbox#30]
Some users are reporting issues like this 

```
2016-12-09T12:50:37      1             Traceback (most recent call last):
                                                 File "/home/qgis/.qgis3/python/plugins/gml_application_schema_toolbox/gui/import_gmlas_panel.py", line 208, in on_validateButton_clicked
                                                   self.validate()
                                                 File "/home/qgis/.qgis3/python/plugins/gml_application_schema_toolbox/gui/import_gmlas_panel.py", line 214, in validate
                                                   data_source = self.gmlas_datasource()
                                                 File "/home/qgis/.qgis3/python/plugins/gml_application_schema_toolbox/gui/import_gmlas_panel.py", line 202, in gmlas_datasource
                                                   open_options=openOptions)
                                                 File "/usr/lib/python3/dist-packages/osgeo/gdal.py", line 2995, in OpenEx
                                                   return _gdal.OpenEx(*args, **kwargs)
                                               RuntimeError: Cannot resolve http://inspire.ec.europa.eu/schemas/hh/4.0/HumanHealth.xsd

```

and it sounds related to proxy. Proxy can be set in QGIS and I suppose user have to also set it at OGR level as it looks it does not make any differences. 

There is 2 proxy config:

* one at GDAL level (GDAL_HTTP_PROXY,  GDAL_PROXY_AUTH, GDAL_HTTP_PROXYUSERPW

* one at XLink resolution level  https://github.com/rouault/gdal2/blob/gmlas/gdal/data/gmlasconf.xsd#L334

3 questions:

* Do we have to set the QGIS proxy options to OGR for the conversion ? 
* How to set it  ? env variables or API 
* Do we have to set both (ie. GDAL options and xlinks) ?

**Comment by rouault** [comment]
> Do we have to set the QGIS proxy options to OGR for the conversion ?

OGR isn't aware of QGIS proxy settings, so yes it might make sense to "translate" QGIS proxy settings to GDAL.

> How to set it ? env variables or API

Both are possible. GDAL configuration options are read from the values set with gdal.SetConfigOption(key, value), or if not set, it will retrieve them from the environment variable of name 'key'.

> Do we have to set both (ie. GDAL options and xlinks) ?

No, just setting it as GDAL config options will be used for both schemas and xlink resolution. Actually I'm not sure why I put an option specific to xlink. Proxying issues indeed are the same for schema downloading or xlink resolution.

**Comment by arnaud-morvan** [comment]
IMHO, It seems it make sense to make this at QGIS core level.

**Comment by frcaldeira** [comment]
Hello all,
I Noticed that in System Setting  -> Network -> proxy I could define the proxy server settings for the Ubuntu OS, works for Firefox and Gdal. So when I tried the command ogrinfo GMLAS:CidadesEstatisticas2014_CONT_3.gml got the fallowing message

Seems to be ok. Actually close enough comparing to the samples you provided.

qgis@qgis-VirtualBox:~/Documents$ ogrinfo GMLAS:CidadesEstatisticas2014_CONT_3.gml
Had to open data source read-only.
INFO: Open of `GMLAS:CidadesEstatisticas2014_CONT_3.gml'
      using driver `GMLAS' successful.
1: areastatisticalunit (None)
2: areastatisticalunit_metadataproperty (None)
3: areastatisticalunit_name (None)
4: areastatisticalunit_administrativeunit (None)
5: areastatisticalunit_lowers (None)
6: areastatisticalunit_uppers (None)
7: areastatisticalunit_successors (None)
8: areastatisticalunit_predecessors (None)
9: evolution (None)
10: evolution_metadataproperty (None)
11: evolution_name (None)
12: evolution_finalunitversions (None)
13: evolution_units (None)
14: evolution_initialunitversions (None)
15: vectorstatisticalunit (None)
16: vectorstatisticalunit_metadataproperty (None)
17: vectorstatisticalunit_name (None)
18: statisticaltessellation (None)
19: statisticaltessellation_metadataproperty (None)
20: statisticaltessellation_name (None)
21: statisticaltessellation_units (None)
22: location
23: timeinstant (None)
24: timeinstant_metadataproperty (None)
25: timeinstant_name (None)
26: timeinstant_relatedtime (None)
27: timeperiod (None)
28: timeperiod_metadataproperty (None)
29: timeperiod_name (None)
30: timeperiod_relatedtime (None)
31: geographicalname1 (None)
32: geographicalname_spelling (None)
33: condominium (Multi Surface)
34: condominium_metadataproperty (None)
35: condominium_gml_name (None)
36: condominium_au_name (None)
37: condominium_admunit (None)
38: administrativeunit (Multi Surface)
39: administrativeunit_metadataproperty (None)
40: administrativeunit_gml_name (None)
41: administrativeunit_nationallevelname (None)
42: administrativeunit_au_name (None)
43: administrativeunit_residenceofauthority (Point)
44: administrativeunit_condominium (None)
45: administrativeunit_lowerlevelunit (None)
46: administrativeunit_administeredby (None)
47: administrativeunit_coadminister (None)
48: administrativeunit_boundary (None)
49: administrativeboundary (Curve)
50: administrativeboundary_metadataproperty (None)
51: administrativeboundary_name (None)
52: administrativeboundary_nationallevel (None)
53: administrativeboundary_admunit (None)
54: inspireid (None)
55: thematicid (None)
56: geographicalname2 (None)
57: validityperiod (None)
58: referenceperiod (None)
59: geometry
60: evolutions (None)
61: areastatisticalunit_thematicid_thematicid (None)
62: areastatisticalunit_geographicalname_geographicalname (None)
63: areastatisticalunit_geometry_geometry (None)
64: areastatisticalunit_evolutions_evolutions (None)
65: vectorstatisticalunit_thematicid_thematicid (None)
66: vectorstatisticalunit_geographicalname_geographicalname (None)
67: vectorstatisticalunit_geometry_geometry (None)
68: vectorstatisticalunit_evolutions_evolutions (None)

Regards
Francisco

**Comment by arnaud-morvan** [comment]
@frcaldeira : Yes, if you put your proxy setting at system level, it should work, but this is about using the proxy defined in QGIS settings.

**Comment by frcaldeira** [comment]
Hello all,

Ok Arnaud-Morvan I understand.

I confirm that using the settings of QGIS in the menu **Settings** -> **options** -> **network**, and option **Use proxy for web access** (with definition of proxy type, host, user, password) is not enough for OGR conversion.

Regards
F
